projects/ad2s1210_iio/app/ad2s1210_iio.c

```diff
@@ -0,0 +1,619 @@
+						  * iio_dev_data->buffer->bytes_per_scan;
+		buf_size_updated = true;
+	}
+
+	/* Push into circular buffer */
+	for (i = 0; i < RESOLVER_CHANNELS; i++) {
+		if (ad2s1210_iio_channels[i].ch_type == IIO_ANGL)
+			chn = AD2S1210_POS;
+		else
+			chn = AD2S1210_VEL;
+
+		if (!(iio_dev_data->buffer->active_mask & (1 << i)))
+			continue;
+
+		ret = ad2s1210_spi_single_conversion(ad2s1210_dev_inst,
+						     chn, &data);
+		if (ret)
+			return ret;
+
+		ret = no_os_cb_write(iio_dev_data->buffer->buf, &data,
+				     BYTES_PER_SAMPLE);
+		if (ret)
+			return ret;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief	Init for reading/writing and parameterization of a
+ *			ad2s1210 IIO device
+ * @param	desc[in,out] - IIO device descriptor
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t ad2s1210_iio_param_init(struct iio_device **desc)
+{
+	struct iio_device *iio_ad2s1210_inst;
+
+	iio_ad2s1210_inst = calloc(1, sizeof(struct iio_device));
+	if (!iio_ad2s1210_inst)
+		return -ENOMEM;
+
+	iio_ad2s1210_inst->num_ch = NO_OS_ARRAY_SIZE(ad2s1210_iio_channels);
+	iio_ad2s1210_inst->channels = ad2s1210_iio_channels;
+	iio_ad2s1210_inst->attributes = ad2s1210_iio_global_attributes;
+
+	iio_ad2s1210_inst->submit = iio_ad2s1210_submit_buffer;
+	iio_ad2s1210_inst->pre_enable = iio_ad2s1210_prepare_transfer;
+	iio_ad2s1210_inst->post_disable = iio_ad2s1210_end_transfer;
+	iio_ad2s1210_inst->debug_reg_read = iio_ad2s1210_debug_reg_read;
+	iio_ad2s1210_inst->debug_reg_write = iio_ad2s1210_debug_reg_write;
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+	iio_ad2s1210_inst->trigger_handler = ad2s1210_trigger_handler;
+#endif
+
+	*desc = iio_ad2s1210_inst;
+
+	return 0;
+}
+
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+/**
+ * @brief       Initialization of AD2S1210 IIO hardware trigger parameters
+ * @param       desc[in,out] - IIO hardware trigger descriptor
+ * @return      0 in case of success, negative error code otherwise
+ */
+static int32_t ad2s1210_iio_trigger_param_init(struct iio_hw_trig **desc)
+{
+	int32_t ret;
+	struct iio_hw_trig_init_param ad2s1210_hw_trig_init_params;
+	struct iio_hw_trig *hw_trig_desc;
+
+	hw_trig_desc = calloc(1, sizeof(struct iio_hw_trig));
+	if (!hw_trig_desc)
+		return -ENOMEM;
+
+	ad2s1210_hw_trig_init_params.irq_id = TRIGGER_INT_ID;
+	ad2s1210_hw_trig_init_params.name = AD2S1210_IIO_TRIGGER_NAME;
+	ad2s1210_hw_trig_init_params.irq_trig_lvl = NO_OS_IRQ_EDGE_FALLING;
+	ad2s1210_hw_trig_init_params.irq_ctrl = trigger_irq_desc;
+	ad2s1210_hw_trig_init_params.cb_info.event = NO_OS_EVT_GPIO;
+	ad2s1210_hw_trig_init_params.cb_info.peripheral = NO_OS_GPIO_IRQ;
+	ad2s1210_hw_trig_init_params.cb_info.handle = trigger_gpio_handle;
+	ad2s1210_hw_trig_init_params.iio_desc = ad2s1210_iio_desc;
+
+	/* Initialize hardware trigger */
+	ret = iio_hw_trig_init(&hw_trig_desc, &ad2s1210_hw_trig_init_params);
+	if (ret)
+		return ret;
+
+	*desc = hw_trig_desc;
+
+	return 0;
+}
+#endif
+/**
```

Comment:
A blank line can be added between two functions

Reply:
ok